LuatixHQ/opencti
# [frontend/backend] - Add default values for mandatory fields to SDO, relationships and sightings

### Proposed changes

* Add default values mechanism for entity setting -> Rework of entity setting screen
* Add default marking definitions by group to be used in default values mechanism -> Added in group edition
* Clean default marking definitions to keep only the top marking by definition type (could be generalize)
* Add new hook to use default values in entities creation and use it in all form creation
* Move static default values in form creation to useDefaultValues hook (ex: confidence)
* Add custom label for CreatedByField & ObjectAssigneeField
* Improve delete inner relation methods to take multiple TO (back-end)

* Raw values changed
    * 4 create/delete events -> in schema-validator test, a data component is created then removed multiple time
    * 1 update event -> in schema-validator test, the entity setting

### Code review without too much pain

* All creation forms have the new mechanism, so it's better to spend more time on the other files that really contain the logic

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/3211

### How to test

* Go to an entity setting
* Add some default values on fields (try diferent types -> multiple string, scale, ov)
* Reload page
* Go to the specific creation form and view your default values (let or remove them)
* Create the entity and check if everything is correct

### Checklist







### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...
